modules/flight_interface/flight_interface.py

```diff
@@ -46,6 +62,15 @@ def __init__(
 
         self.controller = controller
         self.home_location = home_location
+        self.first_waypoint = first_waypoint
+        self.first_waypoint_distance_tolerance = first_waypoint_distance_tolerance
+        self.__run = False
+
+    def __distance_to_first_waypoint(self, global_position: drone_odometry.DronePosition) -> float:
+        delta_x = global_position.latitude - self.first_waypoint.latitude
+        delta_y = global_position.longitude - self.first_waypoint.longitude
+        delta_z = global_position.altitude - self.first_waypoint.altitude
+        return math.sqrt(delta_x**2 + delta_y**2 + delta_z**2)
```

Comment:
minor optimization, but using tolerance squared and not square rooting here would be faster (square roots are expensive) it also reduces dependencies so we don't have to import math